packages/zowe-explorer/__tests__/__e2e__/step_definitions/utils/datasetExtensions.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+/**
+ * Local copy of dataset extension detection logic for e2e tests.
+ * Kept in tests to avoid importing vscode through the API dependency chain.
+ */
+export const DS_EXTENSION_MAP: Map<string, (string | RegExp)[]> = new Map([
```

Comment:
I don't like defining this map twice in our codebase very much, but I guess we don't have an easy alternative right now. Thanks for adding the comment to explain why.

Reply:
Yeah it feels really hacky to duplicate it 😓 we can investigate separating exports down the road (that don't rely on VS Code) so the import isn't encountered through the require stack.